parla/task_runtime.py

```diff
@@ -291,32 +269,40 @@ def __repr__(self):
 
 
 class Task:
+    __slots__ = [
+        '_mutex', '_name', '_taskid', '_state', '_successors', '_predecessors',
+        '_req', '_assigned', '_dependent_events', '_num_blocking_predecessors',
+        '__dict__'
+    ]
+
     def __init__(self, dependencies: Collection["Task"], taskid,
-                 req: ResourceRequirements, name: Optional[str] = None):
+                 req: ResourceRequirements, name: Optional[str] = None,
+                 init_state: TaskState = TaskWaiting(),
+                 init_assigned: bool = False):
         self._mutex = threading.Lock()
         with self._mutex:
             # This is the name of the task, which is distinct from the TaskID. It inherits its name from the func.
             self._name = name
-            self._dependees = []
+            self._successors: List[Task] = []
             self._taskid = taskid
+            self._state = init_state
             # Maintain dependencies as a list object.
             # Therefore, bi-directional edges exist among dependent tasks.
             # Some of these dependencies are moved to a data movement task.
-            self._set_dependencies(dependencies)
+            self.predecessors = dependencies
             self._req = req
             # This flag specifies if a task is assigned device.
             # If it is, it sets to True. Otherwise, it sets to False.
-            self.assigned = False
-            # Dependent tasks' event objects for runahead scheduling.
-            self.dependent_events = []
-            self.events = []
+            self._assigned = init_assigned
+            # Predecessor tasks' event objects for runahead scheduling.
+            self._predecessor_events = []
 
     @property
-    def taskid(self) -> TaskID:
+    def taskid(self) -> 'TaskID':
```

Comment:
What does this change do?

Reply:
To avoid circular import. It's a Python feature.